src/main/java/com/rikaisan/mixin/BlockLogicWireRedstoneMixin.java

```diff
@@ -9,63 +14,31 @@
-	 */
-	@Overwrite
-	public boolean getSignal(WorldSource worldSource, int x, int y, int z, Side side) {
-		if (!this.shouldSignal || worldSource.getBlockMetadata(x, y, z) == 0) {
-			return false;
-		} else if (side == Side.TOP) {
-			return true;
-		} else {
-			boolean negXShouldConnectTo = shouldConnectTo(worldSource, x - 1, y, z, 1)
-				|| !worldSource.isBlockNormalCube(x - 1, y, z) && AdditionalRedstoneWireLogic.shouldConnectToDiagonal(worldSource, x - 1, y - 1, z, -1, Side.TOP);
-			boolean posXShouldConnectTo = shouldConnectTo(worldSource, x + 1, y, z, 3)
-				|| !worldSource.isBlockNormalCube(x + 1, y, z) && AdditionalRedstoneWireLogic.shouldConnectToDiagonal(worldSource, x + 1, y - 1, z, -1, Side.TOP);
-			boolean negZShouldConnectTo = shouldConnectTo(worldSource, x, y, z - 1, 2)
-				|| !worldSource.isBlockNormalCube(x, y, z - 1) && AdditionalRedstoneWireLogic.shouldConnectToDiagonal(worldSource, x, y - 1, z - 1, -1, Side.TOP);
-			boolean posZShouldConnectTo = shouldConnectTo(worldSource, x, y, z + 1, 0)
-				|| !worldSource.isBlockNormalCube(x, y, z + 1) && AdditionalRedstoneWireLogic.shouldConnectToDiagonal(worldSource, x, y - 1, z + 1, -1, Side.TOP);
-
-			if (!worldSource.isBlockNormalCube(x, y + 1, z)) {
-				if (worldSource.isBlockNormalCube(x - 1, y, z) && AdditionalRedstoneWireLogic.shouldConnectToDiagonal(worldSource, x - 1, y + 1, z, -1, Side.BOTTOM)) {
-					negXShouldConnectTo = true;
-				}
 
-				if (worldSource.isBlockNormalCube(x + 1, y, z) && AdditionalRedstoneWireLogic.shouldConnectToDiagonal(worldSource, x + 1, y + 1, z, -1, Side.BOTTOM)) {
-					posXShouldConnectTo = true;
-				}
 
-				if (worldSource.isBlockNormalCube(x, y, z - 1) && AdditionalRedstoneWireLogic.shouldConnectToDiagonal(worldSource, x, y + 1, z - 1, -1, Side.BOTTOM)) {
-					negZShouldConnectTo = true;
-				}
-
-				if (worldSource.isBlockNormalCube(x, y, z + 1) && AdditionalRedstoneWireLogic.shouldConnectToDiagonal(worldSource, x, y + 1, z + 1, -1, Side.BOTTOM)) {
-					posZShouldConnectTo = true;
-				}
-			}
+	@WrapOperation(method = "getSignal", at = @At(value = "INVOKE", target = "Lnet/minecraft/core/block/BlockLogicWireRedstone;shouldConnectTo(Lnet/minecraft/core/world/WorldSource;IIII)Z"))
+	private boolean fixDiagonal(WorldSource worldSource, int x, int y, int z, int data, Operation<Boolean> original, @Local(name = "y") int originalY) {
+		if(y == originalY - 1) return AdditionalRedstoneWireLogic.shouldConnectToDiagonal(worldSource, x, y, z, data, Side.TOP);
+		if(y == originalY + 1) return AdditionalRedstoneWireLogic.shouldConnectToDiagonal(worldSource, x, y, z, data, Side.BOTTOM);
+		return original.call(worldSource, x, y, z, data);
+	}
 
+	@Definition(id = "negZShouldConnectTo", local = @Local(type = boolean.class, name = "negZShouldConnectTo"))
+	@Expression("negZShouldConnectTo")
+	@Inject(method = "getSignal", at = @At("MIXINEXTRAS:EXPRESSION"), cancellable = true)
```

Comment:
Add `ordinal` to make sure the inject only happens at the first occurrence.

Fixes https://github.com/Rikaisan/BTARedstoneTweaks/pull/2#pullrequestreview-2588696934

Reply:
Fixed by https://github.com/Rikaisan/BTARedstoneTweaks/pull/2/commits/d70a0a3a3866270eaf556390cfc0d08f686d34c2